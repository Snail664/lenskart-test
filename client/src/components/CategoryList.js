// node modules
import React from "react";
import { Link } from "react-router-dom";

// misc
import uniqPropFromArrOfObj from "../utils/uniqPropFromArrOfObj";

const CategoryList = ({ products }) => {
  // use helper function to return array of uniqe categories
  const renderCategories = uniqPropFromArrOfObj(products, "category").map(
    (category, index) => {
      // map over returned array and render item
      return (
        <div className="p-5 m-5 w-100 h-100 bg-dark rounded" key={index}>
          <Link to={`/categories/${category}`}>
            <h2 className="text-white">{category}</h2>
          </Link>
        </div>
      );
    }
  );
  return (
    <div className="text-center">
      <h1>Category List</h1>
      <h4>Please select a category</h4>
      <div className="d-flex flex-wrap justify-content-center">
        {renderCategories}
      </div>
    </div>
  );
};

export default CategoryList;
