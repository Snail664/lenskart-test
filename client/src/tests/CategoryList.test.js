import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Router } from "react-router-dom";
import history from "../history";

import CategoryList from "../components/CategoryList";
import { dummyData } from "./dummyData";

// Category renders correctly
it("Matches snapshot for loaded data", () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <CategoryList products={dummyData} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
