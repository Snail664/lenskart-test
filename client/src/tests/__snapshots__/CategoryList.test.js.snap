// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot for loaded data 1`] = `
<div
  className="text-center"
>
  <h1>
    Category List
  </h1>
  <h4>
    Please select a category
  </h4>
  <div
    className="d-flex flex-wrap justify-content-center"
  >
    <div
      className="p-5 m-5 w-100 h-100 bg-dark rounded"
    >
      <a
        href="/categories/spectacles"
        onClick={[Function]}
      >
        <h2
          className="text-white"
        >
          spectacles
        </h2>
      </a>
    </div>
    <div
      className="p-5 m-5 w-100 h-100 bg-dark rounded"
    >
      <a
        href="/categories/sunglasses"
        onClick={[Function]}
      >
        <h2
          className="text-white"
        >
          sunglasses
        </h2>
      </a>
    </div>
  </div>
</div>
`;
